lib/timeutils/strptime-tz.c

```diff
@@ -349,6 +351,35 @@ strptime_with_tz(const char *buf, const char *fmt, struct tm *tm, long *tm_gmtof
           state |= S_MDAY;
           continue;
 
+        case 'f':
+        {
+          const unsigned char *end = conv_num(bp, tm_usec, 0, 999999);
+          int digits = end - bp;
+
+          /*
+           * We have read up to 6 digits, but if the message has
+           * sub-microsecond precision, eat-up the digits we cannot handle.
+           */
+          while (isdigit(*end))
+            {
+              end++;
+            }
+
+          /*
+           * If we read less than 6 digits, we need to adjust the value:
+           * "012" was parsed as 12 but is 12000 us.
+           */
+          while (digits++ < 6)
+            {
+              *tm_usec *= 10;
+            }
+
+          bp = end;
+          LEGAL_ALT(ALT_O);
```

Comment:
I am unfamiliar with these `LEGAL_ALT`, `ALT_O` concepts. What is this line used for?

Reply:
D'oh, good catch, it's a mistake!  According to `strftime(3)`:

```
     %E* %O*                                                                    
           POSIX locale extensions.  The sequences %Ec %EC %Ex %EX %Ey %EY %Od  
           %Oe %OH %OI %Om %OM %OS %Ou %OU %OV %Ow %OW %Oy are supposed to      
           provide alternate representations.                                   

           Additionally %OB implemented to represent alternative months names
           (used standalone, without day mentioned).
```

`%Of` does not mean anything, and I should have written `LEGAL_AT(0)` to ensure we are _not_ parsing `%Of` nor `%Ef`…  I'll amend the commit accordingly.

Reply:
Done, thanks!